### PR TITLE
pow: add access to pre-digest for algorithm verifiers

### PR DESCRIPTION
**Pull request by sorpaas**
This adds access for verifiers to the specific POW pre-digest. The pre-digest may encode information such as the pow author, and is used for verification such as signed mining.

### CODE REVIEW

File: client/consensus/pow/src/lib.rs [hunk 1]
```diff
@@ -678,3 +686,22 @@ fn mine_loop<B: BlockT, C, Algorithm, E, SO, S, CAW>(
...
+	for log in header.digest().logs() {
+		trace!(target: "pow", "Checking log {:?}, looking for pre runtime digest", log);
+		match (log, pre_digest.is_some()) {
+			(DigestItem::PreRuntime(POW_ENGINE_ID, _), true) => {
+				return Err(Error::MultiplePreRuntimeDigests)
+			},
+			(DigestItem::PreRuntime(POW_ENGINE_ID, v), false) => {
+				pre_digest = Some(v.clone());
+			},
+			(_, _) => trace!(target: "babe", "Ignoring digest not meant for us"),
```
Comment (bkchr) [inline client/consensus/pow/src/lib.rs:702]:
```suggestion
			(_, _) => trace!(target: "pow", "Ignoring digest not meant for us"),
```

File: client/consensus/pow/src/lib.rs [hunk 2]
```diff
@@ -505,7 +512,7 @@ pub fn start_mine<B: BlockT, C, Algorithm, E, SO, S, CAW>(
 	client: Arc<C>,
 	algorithm: Algorithm,
 	mut env: E,
-	preruntime: Option<Vec<u8>>,
+	pre_runtime: Option<Vec<u8>>,
```
Comment (bkchr) [inline client/consensus/pow/src/lib.rs:515]: This should be renamed in the docs above as well.

File: client/consensus/pow/src/lib.rs (hunk 1 above)
Comment (sorpaas) [reply client/consensus/pow/src/lib.rs:702]: Ah sorry!

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge

**Comment by sorpaas** [comment]
bot merge

**Pull request merged**